src/root.jsx

```diff
@@ -5,45 +5,23 @@ import DataTable from "./data-table/index.jsx";
 import { LinkedScrollWrapper, LinkedScrollSection } from "./linked-scroll";
 
 class Root extends React.PureComponent {
-  constructor (props) {
-    super(props);
-    this.onDataTableRefSet = this.onDataTableRefSet.bind(this);
-    this.renderedTableWidth = 0;
-  }
-
-  componentDidUpdate () {
-    let tableWidth;
-    if (this.dataTableRef) {
-      tableWidth = this.dataTableRef.getBoundingClientRect().width;
-      if (this.renderedTableWidth !== tableWidth) {
-        this.forceUpdate();
-      }
-    }
-  }
-
-  onDataTableRefSet (element) {
-    this.dataTableRef = element;
-    this.forceUpdate();
-  }
-
   render () {
     const { editmodeClass, component, state } = this.props;
-    const { data, general, styling, error } = state;
+    const { data, general, styling, error, element } = state;
 
     // Determine cell- and column separator width
     let cellWidth = '0px';
     let columnSeparatorWidth = '';
-    if (this.dataTableRef && !error) {
-      const tableWidth = this.dataTableRef.getBoundingClientRect().width;
-      this.renderedTableWidth = tableWidth;
+    if (!error && element) {
+      const tableWidth = element.getBoundingClientRect().width;
```

Comment:
This is the big change, instead of using the reference element we use the root for the chart to avoid the infinite loop.